Remove jandex plugin from flaky test

glefloch opened the pull request:
This removes the jandex plugin from a sample test project. On windows, it looks the `jandex` task was running after the `jar` task which is not correct.

glefloch committed:
Remove gradle connection for included build discovery

geoand commented:
Thanks @glefloch.

It seems like the Gradle tests are failing on Windows, is it related?

glefloch commented:
Yes, this test fails on multiple PRs and I try to fix it. But it looks like my fix isn't enough. I will keep working on it.

glefloch commented:
@aloubyansky I updated the include build discovery process and removed the usage of the `GradleConnector`. 
By looking for the `QuarkusModel` all substitution needed to have the `quarkus` plugin enabled which should not be the case as we can substitute any library. 
As Gradle always output resources in `build/resources/main` and classes in `build/classes/java/main` or `build/classes/kotlin/main`, we can easily build those paths from the project path.
Also on windows, there was a weird behavior regarding task ordering, so this should also fix it. WDYT?

gastaldi commented on devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java:
```diff
@@ -413,13 +415,17 @@ private void addDevModePaths(final DependencyImpl dep, ResolvedArtifact a, Proje
+    private void addSubstitutedProject(final DependencyImpl dep, File projectFile) {
+        dep.addPath(new File(projectFile, MAIN_RESOURCES_OUTPUT));
+        File classesOutput = new File(projectFile, CLASSES_OUTPUT);
+        for (File languageDirectory : classesOutput.listFiles()) {
```
`listFiles` may return null if `classesOutput` isn't a valid directory

glefloch replied:
Oh yes, I will add a check on this one.

aloubyansky commented on devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java:
```diff
@@ -413,13 +415,17 @@ private void addDevModePaths(final DependencyImpl dep, ResolvedArtifact a, Proje
-        for (File sourceDirectory : moduleOutput.getSourceDirectories()) {
-            dep.addPath(sourceDirectory);
+    private void addSubstitutedProject(final DependencyImpl dep, File projectFile) {
+        dep.addPath(new File(projectFile, MAIN_RESOURCES_OUTPUT));
```
Does this path always exist?

glefloch replied:
I will add a check too, we have seen some project with no resource directory.

aloubyansky commented:
That is supposed to cover the default layout, right? It could still be different in some projects?

glefloch commented:
@aloubyansky, the output layout should not differ a lot. Usually, there is custom layout for tests.
So yes, if projects use something else that `src/main/java`, `src/main/kotlin` or `src/main/scala` for sources, then it won't be handled but I think most project respect `src/main` for project sources. 
If we encounter another layout, we could load a `built-in` gradle model to get more infos about the project and handled that.

aloubyansky commented:
Makes sense, thanks!

gsmet commented:
CI looks good (at least on the Gradle front) but I'm not really qualified to review this PR.

@aloubyansky if everything is OK for you, I let you approve and merge?
